esp-hal-procmacros/src/lib.rs

```diff
@@ -437,6 +437,66 @@ pub fn interrupt(args: TokenStream, input: TokenStream) -> TokenStream {
     .into()
 }
 
+#[cfg(feature = "interrupt")]
+#[proc_macro_attribute]
+pub fn handler(args: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
Do we need this macro? Won't the type system i.e the set_interrupt_handler function how handle ensuring that the type is of `extern "C"` with no arguments?

Reply:
Yes, this is just sugar to hide the `extern "C"`